### ISSUE TITLE
Add search

### ISSUE THREAD

**Issue by jace** [hasgeek/funnel#438]
Funnel needs the following three search endpoints, all of which accept a `q` parameter for the query and a `type` parameter for the object type.

1. `/search`
2. `/<profile>/search`
3. `/<profile>/<project>/search`

Following the convention adopted in #321 (URL structure), the word `search` now joins the reserved names list.

Search will apply to the following types:

1. Profile (except when searching in a profile or project)
2. Project (except when searching in a project)
3. Proposal
4. Session
5. Comment
6. Label

Users are not included as there is nowhere to link to for users without profiles. Users will start appearing in search results once user profile pages are enabled (in #292). 

The search page shows tabs for each type of result, with a pill containing the result count. Results are sorted by relevance (words appearing in titles take higher priority than content body, etc) and batched 20 at a time, with additional batches loaded automatically as the user scrolls down. There will be no batch navigation buttons.

The backend will accept a `start` parameter for the row number to begin a new batch at.

The front-end will stop attempting to load batches once the originally promised number of results have been loaded.

**Comment by jace** [comment]
First cut in #438.

**Comment by iambibhas** [comment]
Should we also figure out how to track the searches and the popular resources that get consumed?

**Comment by jace** [comment]
Yes, but first cut can be from GA. Let's build custom analytics when we find something GA can't help with.

**Comment by jace** [comment]
Profile and project search were added a while ago, but the UI to access them is pending. We also need two other features:

1. Autocomplete for search results. Don't wait for the user to submit.
2. Consolidated results from across document types, with pills to let the user pick one/many (none selected implies all selected)

**Comment by jace** [comment]
Project and profile search is coming to the UI with #961. Label search is currently excluded because labels as a product feature are underdeveloped. The search feature is effectively complete as per this ticket's specification. A future ticket may want to introduce a DSL that extends the Postgres syntax, so that a query may filter on fields other than `search_vector`.
